client/signup/controller.js

```diff
@@ -363,8 +375,6 @@ export default {
 		const signupDependencies = getSignupDependencyStore( getState() );
 
 		const siteIdOrSlug =
-			query?.site ||
-			signupDependencies?.site ||
 			signupDependencies?.siteSlug ||
```

Comment:
switching positions between `signupDependencies` and `query` can be a better fix to the precedence issue

Reply:
I shied away from that due to concerns about existing signup flows potentially being silently dependent on this behavior. There too many flows and I don't think we can reasonably audit/test them all. Adding something new and moving anything that needs to over to it is the safer change operation.

Reply:
Me too .. but there is a lot to test when doing this change, the affected code scope is too wide gonna force it on the `start` function is a hacky approach but it works